syntax = "proto3";

package execution;

service WorkflowExecution {
  // Trigger a workflow execution
  rpc ExecuteWorkflow(ExecuteWorkflowRequest) returns (ExecuteWorkflowResponse);

  // Get status of a workflow execution
  rpc GetWorkflowStatus(GetWorkflowStatusRequest) returns (WorkflowStatusResponse);
}

message ExecuteWorkflowRequest {
  int32 workflow_id = 1;
}

message ExecuteWorkflowResponse {
  int32 execution_id = 1;
  string status = 2;
}

message GetWorkflowStatusRequest {
  int32 execution_id = 1;
}

message WorkflowStatusResponse {
  int32 execution_id = 1;
  int32 workflow_id = 2;
  string status = 3;
  string started_at = 4;
  string completed_at = 5;
}
